webportal/src/app/cluster-view/services/services.component.js

```diff
@@ -66,3 +66,30 @@ $(document).ready(() => {
 });
 
 module.exports = { loadservices }
+
+// This function will call kubernetes restful api to get node - podlist - label info, to support service view monitor page.
+function getServiceView(kubernetesUrl) {
+  var nodesInfo = []
+  var request = require('sync-request');
+  var resNodes = request('GET', kubernetesUrl + '/api/v1/nodes');
```

Comment:
I tried your code but still got the following error:

> http://10.151.40.179:9090/api/v1/node: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:9286' is therefore not allowed access.

Reply:
var http = require('http');
var sync = require('simplesync');
http.createServer(function (request, response) {
  response.writeHead(200, { 'Content-Type': 'text/html;    charset=utf-8' });
  if (request.url !== "/favicon.ico") {
    sync.block(function () {
      var res = getServiceView("http://10.151.40.222:8080")
      console.log(JSON.stringify(res));
      response.write(JSON.stringify(res));
      response.end();
    })
  }
}).listen(8000);

console.log('Server running at http://127.0.0.1:8000/');

function getServiceView(kubernetesUrl) {
  var nodesInfo = []
  var request = require('sync-request');
  var resNodes = request('GET', kubernetesUrl + '/api/v1/nodes');
  var data = JSON.parse(resNodes.body.toString('utf-8'));
  var items = data.items;
  for (var i in items) {
    var node = items[i].metadata.name
    console.log(node)
    var resSingleNode = request('GET', kubernetesUrl + '/api/v1/namespaces/kube-system/services/kubernetes-dashboard/proxy/api/v1/node/' + node);
    var nodeinfo = JSON.parse(resSingleNode.body.toString('utf-8'))
    var pods = nodeinfo.podList.pods
    var podList = []
    for (var i in pods) {
      podList.push({ "pod": pods[i].objectMeta.name, "status": pods[i].podStatus.status })
    }
    var labelList = []
    var labels = nodeinfo.objectMeta.labels
    for (var i in labels) {
      labelList.push({ "label": labels[i] })
    }
    var singleNodeInfo = { "node": node, "podList": podList, "labelList": labelList }
    nodesInfo.push(singleNodeInfo)
  }
  return nodesInfo;
}

Reply:
Hi Hao, i use this nodejs code to do the unit test. you could try

Reply:
Is it server-side function? Then how can it be used at client (browser) side?